File: compiler/testData/loadJava/kotlinSignature/propagation/return/InheritReadOnlinessSameClass.java
package test;

import org.jetbrains.annotations.NotNull;
import java.util.List;
import java.util.Collection;

import jet.runtime.typeinfo.KotlinSignature;

public interface InheritReadOnlinessSameClass {

    public interface Super {
        @KotlinSignature("fun foo(): List<String>")
        List<String> foo();
    }

    public interface Sub extends Super {
        List<String> foo();
    }
}

File: compiler/testData/loadJava/kotlinSignature/propagation/return/InheritReadOnlinessSameClass.kt
package test

import org.jetbrains.annotations.NotNull

public open class InheritReadOnlinessSameClass : java.lang.Object() {
    public open fun foo(): List<String> = throw UnsupportedOperationException()

    public open class Sub: InheritReadOnlinessSameClass() {
        override fun foo(): List<String> = throw UnsupportedOperationException()
    }
}
